backend/pick-git/src/main/java/com/woowacourse/pickgit/user/application/UserService.java

```diff
@@ -37,84 +44,38 @@ public UserProfileResponseDto getMyUserProfile(AuthUserRequestDto requestDto) {
             .location(user.getLocation())
             .website(user.getWebsite())
             .twitter(user.getTwitter())
-            .following(false)
+            .following(following)
             .build();
     }
 
     @Transactional(readOnly = true)
-    public UserProfileResponseDto getUserProfile(AppUser user, String targetName) {
+    public UserProfileResponseDto getUserProfile(AuthUserRequestDto requestDto, String targetName) {
         User target = findUserByName(targetName);
-
-        if (user.isGuest()) {
-            return UserProfileResponseDto.builder()
-                .name(target.getName())
-                .imageUrl(target.getImage())
-                .description(target.getDescription())
-                .followerCount(target.getFollowerCount())
-                .followingCount(target.getFollowingCount())
-                .postCount(target.getPostCount())
-                .githubUrl(target.getGithubUrl())
-                .company(target.getCompany())
-                .location(target.getLocation())
-                .website(target.getWebsite())
-                .twitter(target.getTwitter())
-                .following(null)
-                .build();
+        if (requestDto.isGuest()) {
+            return generateUserProfileResponse(target, null);
         }
-
-        User source = findUserByName(user.getUsername());
-
-        return UserProfileResponseDto.builder()
-            .name(target.getName())
-            .imageUrl(target.getImage())
-            .description(target.getDescription())
-            .followerCount(target.getFollowerCount())
-            .followingCount(target.getFollowingCount())
-            .postCount(target.getPostCount())
-            .githubUrl(target.getGithubUrl())
-            .company(target.getCompany())
-            .location(target.getLocation())
-            .website(target.getWebsite())
-            .twitter(target.getTwitter())
-            .following(source.isFollowing(target))
-            .build();
+        User source = findUserByName(requestDto.getGithubName());
+        return generateUserProfileResponse(target, source.isFollowing(target));
     }
 
     public FollowResponseDto followUser(AuthUserRequestDto requestDto, String targetName) {
         User source = findUserByName(requestDto.getGithubName());
         User target = findUserByName(targetName);
-
-        validateDifferentSourceTarget(source, target);
         source.follow(target);
+        return generateFollowResponse(target, true);
+    }
 
+    private FollowResponseDto generateFollowResponse(User target, boolean isFollowing) {
         return FollowResponseDto.builder()
             .followerCount(target.getFollowerCount())
-            .isFollowing(true)
+            .isFollowing(isFollowing)
             .build();
     }
 
     public FollowResponseDto unfollowUser(AuthUserRequestDto requestDto, String targetName) {
```

Comment:
```suggestion
    public FollowResponseDto unFollowUser(AuthUserRequestDto requestDto, String targetName) {
```
unfollow는 원래 소문자로 쓰는게 국룰인가..? 몰라서..

Reply:
undo, unfollow같은 동사가 있어서 이렇게하는게 맞을듯..?